src/filters.cpp

```diff
@@ -137,85 +109,80 @@ NAN_METHOD(Filters::New) {
                         for (std::uint32_t index = 0; index < propertiesLength; ++index) {
-                            v8::Local<v8::Value> property_value = propertyArray->Get(index);
-                            Nan::Utf8String utf8_value(property_value);
-                            int utf8_len = utf8_value.length();
-                            if (utf8_len > 0) {
-                                values.emplace_back(*utf8_value, static_cast<std::size_t>(utf8_len));
+                            Napi::Value property_value = propertyArray.Get(index);
+                            std::string value = property_value.As<Napi::String>();
+                            if (!value.empty()) {
+                                values.emplace_back(value);
                             }
                         }
                         property.first = list;
                         property.second = values;
-                    } else if (layer_properties->IsBoolean() && layer_properties->IsTrue()) {
+                    } else if (layer_properties.IsBoolean() && layer_properties.As<Napi::Boolean>()) {
                         property.first = all;
                         property.second = {};
                     } else {
-                        Nan::ThrowTypeError("invalid filter value, must be an array or a boolean");
+                        Napi::TypeError::New(Env(), "invalid filter value, must be an array or a boolean").ThrowAsJavaScriptException();
                         return;
                     }
-
-                    std::string source_layer = *v8::String::Utf8Value(layer_key);
-
-                    self->add_filter(std::move(source_layer), std::move(filter), std::move(property), minzoom, maxzoom);
+                    std::string source_layer = layer_key.ToString();
+                    add_filter(std::move(source_layer), std::move(filter), std::move(property), minzoom, maxzoom);
                 }
             }
+        } catch (std::exception const& ex) {
+            Napi::TypeError::New(Env(), ex.what()).ThrowAsJavaScriptException();
         }
-        // LCOV_EXCL_START
-        catch (const std::exception& ex) {
-            return Nan::ThrowTypeError(ex.what());
-        }
-        // LCOV_EXCL_STOP
-
-        info.GetReturnValue().Set(info.This());
     }
 }
 
-NAN_METHOD(Filters::layers) {
-    auto layers = Nan::New<v8::Array>();
+Napi::Value Filters::layers(Napi::CallbackInfo const& info) {
+    Napi::HandleScope scope(info.Env());
+    auto layers = Napi::Array::New(Env());
     std::uint32_t idx = 0;
-    auto* filters = Nan::ObjectWrap::Unwrap<Filters>(info.Holder());
-    for (auto const& lay : filters->filters) {
-        Nan::Set(layers, idx++, Nan::New<v8::String>(lay.first).ToLocalChecked());
+    for (auto const& lay : filters) {
+        layers.Set(idx++, lay.first);
     }
-    info.GetReturnValue().Set(layers);
-}
\ No newline at end of file
+    return layers;
```

Comment:
I think this needs to use EscapableScope otherwise it may crash since `layers` could get GC'd while something else is using it. This is an existing bug I think, but could you fix in this branch @artemp ?

Reply:
2a670c0739a14a4788f0a0635ee6d1d4eb0c829a